pkg/framework/aries/framework.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package aries
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+	errors "golang.org/x/xerrors"
+)
+
+// Framework provides access to clients being managed by the Framework.
+type Framework struct {
+	opts options
+}
+
+type options struct {
+	OutboundTransport transport.OutboundTransport
+}
+
+// Option configures the framework.
+type Option func(opts *options) error
+
+// New initializes the Framework based on the set of options provided.
+func New(opts ...Option) (*Framework, error) {
+	// get the default framework options
+	defOpts, err := defFrameworkOpts()
+	if err != nil {
+		return nil, errors.Errorf("Default option init failed : %w", err)
+	}
+
+	frameworkOpts := options{}
+
+	// generate framework configs from options
+	for _, option := range append(defOpts, opts...) {
+		err := option(&frameworkOpts)
+		if err != nil {
+			return nil, errors.Errorf("Error in option passed to New: %w", err)
+		}
+	}
+
+	// create framework with default configs
+	framework := &Framework{
+		opts: frameworkOpts,
+	}
+
+	return framework, nil
+}
+
+// WithOutboundTransport injects a OutboundTransport interface to the Framework
+func WithOutboundTransport(ot transport.OutboundTransport) Option {
+	return func(opts *options) error {
+		opts.OutboundTransport = ot
+		return nil
+	}
+}
+
+// OutboundTransport provides configured OutboundTransport
+func (f *Framework) OutboundTransport() transport.OutboundTransport {
```

Comment:
Why does this need to be exposed? It should be injected into instantiated packages.

Reply:
[DIDExchange Send Request/Response](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/exchange/exchange.go#L90) takes transport has one of the function arguments. The consumer would need to get the transport from framework to pass into this. (Unless we update DID Exchange to use instance of transport).

Reply:
Yes but the Framework should be instantiating a DID exchange instance and supplying a transport to use. The point of a framework is so consumers don't need to manage everything...

Reply:
Note: I don't think that only having *one* transport at a time in the framework works either.

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/e7224c2166bc28f01e32148341d7f003d8fe83c0..226512799fa5cca78500bfcc4f336200df37bea8) based on our discussion (except Context name).